src/server/server.c

```diff
@@ -16,178 +16,166 @@
+        if (b % 2 == 1)
+            result = (result + a) % mod;
+        a = (a * 2) % mod;
+        b /= 2;
+    }
+
+    return result % mod;
 }
 
 uint64_t Factorial(const struct FactorialArgs *args) {
-  uint64_t ans = 1;
+    uint64_t ans = 1;
 
-  // TODO: your code here
+    // TODO: your code here
 
-  return ans;
+    return ans;
 }
 
 void *ThreadFactorial(void *args) {
-  struct FactorialArgs *fargs = (struct FactorialArgs *) args;
-  return (void *) (uint64_t *) Factorial(fargs);
+    struct FactorialArgs *fargs = (struct FactorialArgs *) args;
+    return (void *) (uint64_t *) Factorial(fargs);
 }
 
 int check_tnum(int tnum) {
-  if (tnum <= 0) {
-    printf("tnum must be a positive number\n");
-    return 1;
-  }
-  return 0;
+    if (tnum <= 0) {
+        printf("tnum must be a positive number\n");
+        return 1;
+    }
+    return 0;
 }
 
 int check_port(int port) {
-  if (port < 1024) {
-    printf("port lower than 1024 may not be free\n");
-    return 1;
-  }
-  return 0;
+    if (port < 1024) {
+        printf("port lower than 1024 may not be free\n");
+        return 1;
+    }
+    return 0;
 }
 
-int main(int argc, char **argv) {
-  size_t options_amount = 2;
-
-  static struct option options[] = {{"port", required_argument, 0, 0},
-                                    {"tnum", required_argument, 0, 0},
-                                    {0, 0, 0, 0}};
-
-  //Attach callbacks
-  int (*callbacks[options_amount])(int); //Array of callbacks
-  callbacks[0] = check_port;
-  callbacks[1] = check_tnum;
-
-  int *handled_options = handle_options(argc, argv, options, options_amount, callbacks);
-
-  struct thpool_ *pool = thpool_init(4);
-  thpool_destroy(pool);
-
-  int port = handled_options[0];
-  int tnum = handled_options[1];
-
-  if (port == -1 || tnum == -1) {
-    fprintf(stderr, "Using: %s --port 20001 --tnum 4\n", argv[0]);
-    return 1;
-  }
-
-  int server_fd = socket(AF_INET, SOCK_STREAM, 0);
-  if (server_fd < 0) {
-    fprintf(stderr, "Can not create server socket!");
-    return 1;
-  }
-
-  struct sockaddr_in server;
-  server.sin_family = AF_INET;
-  server.sin_port = htons((uint16_t) port);
-  server.sin_addr.s_addr = htonl(INADDR_ANY);
-
-  int opt_val = 1;
-  setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR, &opt_val, sizeof(opt_val));
-
-  int err = bind(server_fd, (struct sockaddr *) &server, sizeof(server));
-  if (err < 0) {
-    fprintf(stderr, "Can not bind to socket!");
-    return 1;
-  }
-
-  err = listen(server_fd, 128);
-  if (err < 0) {
-    fprintf(stderr, "Could not listen on scket\n");
-    return 1;
-  }
-
-  printf("Server listening at %d\n", port);
-
-  while (true) {
-    struct sockaddr_in client;
-    socklen_t client_len = sizeof(client);
-    int client_fd = accept(server_fd, (struct sockaddr *) &client, &client_len);
-
-    if (client_fd < 0) {
-      fprintf(stderr, "Could not establish new connection\n");
-      continue;
-    }
 
+
+uint64_t naive_factorial(uint64_t n)
```

Comment:
naive?

Reply:
uint64_t n => const uint64_t* n

Reply:
I will rename the function and change the parameter